src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -1140,6 +1140,177 @@ TEST_F(EnsembleFlowCustomNodeFactoryCreateThenExecuteTest, ParallelPipelineFacto
+            if (strcmp(params[i].key, "add_value") == 0) {
+                addValue = atof(params[i].value);
+            }
+            if (strcmp(params[i].key, "sub_value") == 0) {
+                subValue = atof(params[i].value);
+            }
+        }
+
+        *outputsCount = 1;
+        *outputs = internalManager->outputTensor;
+        struct CustomNodeTensor* output = (&(*outputs))[0];
+
+        output->name = "output_numbers";
+        std::cout << input->dataBytes << std::endl;
+        output->data = internalManager->outputTensorData;
+        output->dataBytes = input->dataBytes;
+        output->dims = internalManager->outputTensorDims;
+        output->dimsCount = input->dimsCount;
+        memcpy((void*)output->dims, (void*)input->dims, input->dimsCount * sizeof(uint64_t));
+        output->precision = input->precision;
+
+        for (uint64_t i = 0; i < output->dataBytes; i += sizeof(float)) {
+            *(float*)(output->data + i) = *(float*)(input->data + i) + addValue - subValue;
+        }
+
+        return 0;
+    }
+
+    static int getInputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const struct CustomNodeParam* params, int paramsCount, void* customNodeLibraryInternalManager) {
+        AddSubInternalManager* internalManager = static_cast<AddSubInternalManager*>(customNodeLibraryInternalManager);
+        if (internalManager == nullptr)
+            return 1;
+        *infoCount = 1;
+        *info = internalManager->inputInfo;
+        (*info)->name = "input_numbers";
+        (*info)->dimsCount = 2;
+        (*info)->dims = internalManager->inputDims;
+        (*info)->dims[0] = 1;
+        (*info)->dims[1] = 10;
+        (*info)->precision = FP32;
+        return 0;
+    }
+    static int getOutputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const struct CustomNodeParam* params, int paramsCount, void* customNodeLibraryInternalManager) {
+        AddSubInternalManager* internalManager = static_cast<AddSubInternalManager*>(customNodeLibraryInternalManager);
+        if (internalManager == nullptr)
+            return 1;
+        *infoCount = 1;
+        *info = internalManager->outputInfo;
+        (*info)->name = "output_numbers";
+        (*info)->dimsCount = 2;
+        (*info)->dims = internalManager->outputDims;
+        (*info)->dims[0] = 1;
+        (*info)->dims[1] = 10;
+        (*info)->precision = FP32;
+        return 0;
+    }
+    static int release(void* ptr, void* customNodeLibraryInternalManager) {
+        AddSubInternalManager* internalManager = static_cast<AddSubInternalManager*>(customNodeLibraryInternalManager);
+        if (internalManager == nullptr)
+            return 1;
+        if (!internalManager->isPtrOwnedByManager(ptr)) {
+            free(ptr);
+        }
+        return 0;
+    }
+};
+
+TEST_F(EnsembleFlowCustomNodeFactoryCreateThenExecuteTest, PipelineFactoryCreationAndExecuteWithCustomNodeUsingInternalManager) {
```

Comment:
Can we somehow check that the internal manager was in fact used?

Reply:
We might implement getters and in each getter we could increment `counter` variable? What do you think?

Reply:
What about initializing the buffers with some data and make the execute simply return the data. This way you can check in test if the output data is the same as defined in `initialize`